test/test_native_transfer.py

```diff
@@ -0,0 +1,137 @@
+            """
+            query getByRange {
+                nativeTransfers (
+                filter: {
+                    block: {
+                    timestamp: {
+                        greaterThanOrEqualTo: """ + time_before + """,
+                        lessThanOrEqualTo: """ + time_latest + """
+                        }
+                    }
+                }) {
+                nodes {
+                    denom
+                    toAddress
+                    fromAddress
+                    }
+                }
+            }
+            """
+        )
+
+        # query native transactions, filter by recipient address
+        query_get_by_to_address = gql(
+            """
+            query getByToAddress {
+                nativeTransfers (
+                filter: {
+                    toAddress: {
+                        equalTo: """+json.dumps(self.delegator_address)+"""
+                        }
+                    }
+                ) {
+                nodes {
+                    denom
+                    toAddress
+                    fromAddress
+                    }
+                }
+            }
+            """
+        )
+
+        # query native transactions, filter by sender address
+        query_get_by_from_address = gql(
+            """
+            query getByFromAddress {
+                nativeTransfers (
+                filter: {
+                    fromAddress: {
+                        equalTo: """+json.dumps(self.validator_address)+"""
+                        }
+                    }
+                ) {
+                nodes {
+                    denom
+                    toAddress
+                    fromAddress
+                    }
+                }
+            }
+            """
+        )
+
+        # query native transactions, filter by denomination
+        query_get_by_denom = gql(
+            """
+            query getByDenom {
+                nativeTransfers (
+                filter: {
+                    denom: {
+                        equalTo:\""""+self.denom+"""\"
+                    }
+                }) {
+                    nodes {
+                        denom
+                        toAddress
+                        fromAddress
+                    }
+                }
+            }
+            """
+        )
+
+        queries = [query_get_by_range, query_get_by_to_address, query_get_by_from_address, query_get_by_denom]
+        for query in queries:
+            result = self.gql_client.execute(query)
+            """
+            ["nativeTransfers"]["nodes"][0] denotes the sequence of keys to access the message contents queried for above.
+            This provides {"toAddress":address, "fromAddress":address, "denom":denom, "amount":["amount":amount, "denom":denom]}
+            which can be destructured for the values of interest.
+            """
+            message_ = result["nativeTransfers"]["nodes"]
```

Comment:
Nit: Is there a reason for the underscore suffix? I think it's would be more conventional as `message`, unless that collides with something else.

Reply:
Message is good! Just kept it going out of consistency